src/extension/components/ekubo_oracle.cairo

```diff
@@ -2,30 +2,34 @@ use starknet::ContractAddress;
 
 #[derive(PartialEq, Copy, Drop, Serde, starknet::Store)]
 struct EkuboOracleConfig {
-    oracle_pool: ContractAddress, // Ekubo Oracle pool address
-    quote_token: ContractAddress,
+    decimals: u8,
     period: u64 // [seconds]
 }
 
 #[starknet::component]
 mod ekubo_oracle_component {
     use starknet::ContractAddress;
     use super::EkuboOracleConfig;
+    use vesu::math::pow_10;
     use vesu::units::SCALE;
     use vesu::vendor::{
         ekubo::{
-            IEkuboOracleDispatcher, IEkuboOracleDispatcherTrait, IEkuboCoreDispatcher, IEkuboCoreDispatcherTrait,
-            PoolKey, EKUBO_CORE, MAX_TICK_SPACING
+            construct_oracle_pool_key, IEkuboOracleDispatcher, IEkuboOracleDispatcherTrait, IEkuboCoreDispatcher,
+            IEkuboCoreDispatcherTrait, PoolKey,
         },
-        erc20::{IERC20Dispatcher, IERC20DispatcherTrait}
+        erc20::{ERC20ABIDispatcher, ERC20ABIDispatcherTrait}
```

Comment:
The nittiest of nits (I've only checked because this line changed 😅) - the dispatcher is only used for `decimal()` so it's enough to use the `IERC20Metadata` one. I think it might even so very slightly reduce the contract size 🤓